lib/ansible/modules/windows/win_psmodule.ps1

```diff
@@ -144,15 +267,15 @@ Function Remove-PsModule {
     # If module is present, unistalls it.
     if (Get-Module -Listavailable|?{$_.name -eq $Name}){
       try{
-        Uninstall-Module -Name $Name -Confirm:$false -Force -ErrorAction Stop -WhatIf:$CheckMode | out-null
+        #remove all versions, because now it is possible to have multiple versions installed
+        Uninstall-Module -Name $Name -Confirm:$false -Force -ErrorAction Stop -AllVersions -WhatIf:$CheckMode | out-null
```

Comment:
Should we have a conditional on using `AllVersions` to only remove a particular version if `version` is set?

Reply:
I could, but actually, i do not think it will be usefull. I will try to explain why, if you decide i'm wrong - i will add this conditional.
Simply remove exact version is not enough. Powershell uses latest versions of modules by default, as far as i know. So removing exact version, if it is not the latest will not make any sense (except, maybe, saving some kilobytes). User will need to know which versions are installed already somehow, before he removes particular version to get any effect.

Functionality like `-minimumVersion` or `-maximumVersion` or, maybe, "remove all except" to `absent` will actually duplicate  `present` with `version` parameters in general, but in less straightforward way (it will ensure, that required version is latest on the target host). 
In my opinion, in most cases if one want to remove module he wants to remove it completely.
In case, when one will need to have exact version - `present` with `version` will be enough. It will ensure, required version is latest on the target host.

So waiting for your decision on this one.

Reply:
We should still only remove an individual version if version is set and state=absent. If version is not set then yes all should be removed. There shouldn’t be a reason that I know off for a user to explicitly set a version and state=absent but expect all to be removed.